snapshotter/demux/metrics/proxy.go

```diff
@@ -30,10 +32,71 @@ const (
 	MaxMetricsResponseSize = 32768 // 32 KB
 )
 
+// Monitor is used for port and lifecycle management of metrics proxies.
+type Monitor struct {
+	// monitorChan indicates a port is free for metrics proxy server.
```

Comment:
Slightly confusing for me. Perhaps wording to indicate the channel is used for port de-allocation communication or to free monitor resources.

Reply:
I will update the wording here to make it clearer 👍